core/unix/rseq_linux.c

```diff
@@ -581,7 +581,8 @@ rseq_process_module(module_area_t *ma, bool at_map, bool saw_glibc_rseq_reg)
     ELF_SECTION_HEADER_TYPE *sec_hdr = NULL;
     char *strtab;
     ssize_t load_offs = ma->start - ma->os_data.base_address;
-    if (at_map && elf_hdr->e_shoff + ma->start < ma->end) {
+    if (at_map &&
+        elf_hdr->e_shoff + elf_hdr->e_shnum * sizeof(*sec_hdr) + ma->start < ma->end) {
```

Comment:
Use e_shentsize instead of sizeof?

Reply:
From docx:

```
The product of e_shentsize and e_shnum gives the section header table's size in bytes
```

Reply:
Yes that seems better (note that the assert above would fire if the two were different).  Done.